src/test/java/com/brsanthu/googleanalytics/GoogleAnalyticsApacheHttpTest.java

```diff
@@ -4,27 +4,73 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
+import org.junit.After;
+import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.brsanthu.googleanalytics.request.DefaultRequest;
 import com.brsanthu.googleanalytics.request.GoogleAnalyticsResponse;
 
-public class GoogleAnalyticsTest {
+public class GoogleAnalyticsApacheHttpTest {
+    private static final Logger logger = LoggerFactory.getLogger(GoogleAnalyticsApacheHttpTest.class);
 
-    private static GoogleAnalytics ga = null;
+    protected static GoogleAnalytics ga = null;
+    
+    protected long startTime;
+    protected long callStartTime;
 
     @BeforeClass
     public static void setup() {
-        ga = GoogleAnalytics.builder().withTrackingId(TEST_TRACKING_ID).withAppName("Junit Test").withAppVersion("1.0.0").build();
+        ga = new GoogleAnalyticsApacheHttpTest().getTestBuilder().build();
+    }
+    
+    @Before
+    public void start() {
+        startTime = System.currentTimeMillis();
+    }
+    
+    @After
+    public void end() {
+        logger.debug("Test took: " + (System.currentTimeMillis() - startTime));
+    }
+    
+    protected void startCall() {
+        callStartTime = System.currentTimeMillis();
+    }
+    
+    protected void stopCall() {
+        logger.debug("Call took: " + (System.currentTimeMillis() - callStartTime));;
+    }
+    
+    protected GoogleAnalyticsBuilder getTestBuilder() {
+        return GoogleAnalytics.builder().withTrackingId(TEST_TRACKING_ID).withAppName("Junit ApacheHttp Test").withAppVersion("1.0.0");
     }
```

Comment:
Making the construction of the GoogleAnalyticsBuilder dynamic and overidable allows me to make sure that the OkHttp test gets a GoogleAnalytics instance with the OkHttp client